ui/src/components/view/StoragePoolSelectView.vue

```diff
@@ -0,0 +1,261 @@
+      <div slot="disksizefree" slot-scope="record">
+        <span v-if="record.disksizetotal && record.disksizeused">{{ $bytesToHumanReadableSize(record.disksizetotal * 1 - record.disksizeused * 1) }}</span>
+      </div>
+      <template slot="select" slot-scope="record">
+        <a-radio
+          @click="updateSelection(record)"
+          :checked="selectedStoragePool != null && record.id === selectedStoragePool.id"></a-radio>
+      </template>
+    </a-table>
+    <a-pagination
+      class="top-spaced"
+      size="small"
+      :current="page"
+      :pageSize="pageSize"
+      :total="totalCount"
+      :showTotal="total => `${$t('label.total')} ${total} ${$t('label.items')}`"
+      :pageSizeOptions="['10', '20', '40', '80', '100']"
+      @change="handleChangePage"
+      @showSizeChange="handleChangePageSize"
+      showSizeChanger>
+      <template slot="buildOptionText" slot-scope="props">
+        <span>{{ props.value }} / {{ $t('label.page') }}</span>
+      </template>
+    </a-pagination>
+  </div>
+</template>
+
+<script>
+import { api } from '@/api'
+
+export default {
+  name: 'VolumeStoragePoolSelector',
+  props: {
+    resource: {
+      type: Object,
+      required: true
+    },
+    clusterId: {
+      type: String,
+      required: false,
+      default: null
+    },
+    suitabilityEnabled: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    autoAssignAllowed: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    isOpen: {
+      type: Boolean,
+      required: false
+    }
+  },
+  data () {
+    return {
+      loading: false,
+      storagePools: [],
+      searchQuery: '',
+      totalCount: 0,
+      page: 1,
+      pageSize: 10,
+      selectedStoragePool: null,
+      columns: [
+        {
+          title: this.$t('label.storageid'),
+          scopedSlots: { customRender: 'name' }
+        },
+        {
+          title: this.$t('label.clusterid'),
+          dataIndex: 'clustername'
+        },
+        {
+          title: this.$t('label.podid'),
+          dataIndex: 'podname'
+        },
+        {
+          title: this.$t('label.disksizetotal'),
+          scopedSlots: { customRender: 'disksizetotal' }
+        },
+        {
+          title: this.$t('label.disksizeused'),
+          scopedSlots: { customRender: 'disksizeused' }
+        },
+        {
+          title: this.$t('label.disksizefree'),
+          scopedSlots: { customRender: 'disksizefree' }
+        },
+        {
+          title: this.$t('label.select'),
+          scopedSlots: { customRender: 'select' }
+        }
+      ]
+    }
+  },
+  created () {
+    if (this.suitabilityEnabled) {
+      this.columns.splice(1, 0, { slots: { title: 'suitabilityCustomTitle' }, scopedSlots: { customRender: 'suitability' } }
+      )
+    }
+    this.preselectStoragePool()
+    this.fetchStoragePools()
+  },
+  methods: {
+    fetchStoragePools () {
+      if (this.suitabilityEnabled) {
```

Comment:
Set loading to true / false before / after api call

Reply:
Done